pkgmgmtutils/update_formulas.go

```diff
@@ -0,0 +1,384 @@
+package pkgmgmtutils
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/solo-io/go-utils/githubutils"
+	"github.com/solo-io/go-utils/versionutils"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+)
+
+const PRBaseBranchDefault = "master"
+
+type FormulaOptions struct {
+	Name            string
+	Path            string // repo relative path with file extension
+	RepoOwner       string // repo owner for Formula change
+	RepoName        string // repo name for Formula change
+	PRRepoOwner     string // optional, empty means use RepoOwner
+	PRRepoName      string // optional, empty means use RepoName
+	PRBranch        string // optional, default to master
+	PRDescription   string
+	PRCommitName    string
+	PRCommitEmail   string
+	VersionRegex    string
+	VersionShaRegex string
+	DarwinShaRegex  string
+	LinuxShaRegex   string
+	WindowsShaRegex string
+
+	dryRun bool
+}
+
+type FormulaStatus struct {
+	Name    string
+	Updated bool
+	Err     error
+}
+
+func UpdateFormulas(projectRepoOwner string, projectRepoName string, parentPathSha256 string, fOpts []FormulaOptions) ([]FormulaStatus, error) {
+	//
```

Comment:
nit: Generally we can be more selective and concise with comments. For blocks like this, get rid of the leading and trailing `//` lines. And consider dropping these kinds of comment blocks altogether. I find it just as hard to read this code b/c the functions are so verbose due to the excessive spacing + comments. In the cases where the comments do add something, we could typically achieve the same readability by extracting logic to a helper function.